ledger-api/grpc-definitions/com/daml/ledger/api/v1/active_contracts_service.proto

```diff
@@ -56,9 +58,59 @@ message GetActiveContractsResponse {
 
   // The workflow that created the contracts.
   // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional, In the Ledger API 3.0 this field will be removed.
   string workflow_id = 2;
 
   // The list of contracts that were introduced by the workflow with ``workflow_id`` at the offset.
   // Must be a valid LedgerString (as described in ``value.proto``).
+  // In the Ledger API 3.0 this field will be removed.
+  // This is empty for multi-domain API.
   repeated CreatedEvent active_contracts = 3;
+
+  // Non-empty for multi-domain API.
+  repeated ContractState contract_states = 4;
+}
+
+// In the entire stream of ContractState-s there can be multiple ContractState messages belong to the same contract.
+message ContractState {
+  oneof contract_state {
+    // The contract is active in the given domain. Only one such message is expected per domain.
+    ActiveContract active_contract = 1;
+
+    // This message is only available, if the corresponding transfer-in event cannot be found before or at the
+    // active_at_offset, and for the querying parties it is possible to observe this event after the active_at_offset,
+    // and also to submit the corresponding transfer-in command.
+    // This event typically means, that the transfer is in-flight.
+    IncompleteTransferredOut incomplete_transferred_out = 2;
+
+    // Important: this message is not indicating that the contract is active for the target domain!
+    // This message is only available, if the corresponding transfer-out event cannot be found before or at the
+    // active_at_offset, and for the querying parties it is possible to observe this event after the active_at_offset.
+    // This event typically means, that the the source domain is lagging behind (and the corresponding transfer-out
+    // already happened).
+    TransferredInEvent incomplete_transferred_in = 3;
```

Comment:
As I see from your PR, the submitter, transfer_out_id, and the transfer counter is missing. Do you remember which was hard to get @rgugliel-da ?

Reply:
My initial hope was to serve only what is in https://github.com/DACH-NY/canton/blob/main/community/participant/src/main/scala/com/digitalasset/canton/participant/protocol/transfer/TransferData.scala#L22
But since it was not sufficient, I am querying the event from the event log, so you can have the full event if you want

Reply:
alright let's keep the whole event there